numba/typing/builtins.py

```diff
@@ -866,3 +877,65 @@ class DeferredAttribute(AttributeTemplate):
 
     def generic_resolve(self, deferred, attr):
         return self.context.resolve_getattr(deferred.get(), attr)
+
+
+#------------------------------------------------------------------------------
+
+
+@infer
+class CmpEq(AbstractTemplate):
```

Comment:
The concrete templates for `==` is still needed.  They forces coercion on the args.  This is the generic implementation for all other types.